examples/advanced/pytorch_ddp_example/main.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+import torch
+import torch.distributed as dist
+from omegaconf import DictConfig
+
+import hydra
+
+
+def setup(master_addr: str, master_port: str, rank: int, world_size: int, backend: str):
+    """Initializes distributed process group.
+
+    Arguments:
+        rank: the rank of the current process.
+        world_size: the total number of processes.
+        backend: the backend used for distributed processing.
+    """
+    os.environ["MASTER_ADDR"] = master_addr
+    os.environ["MASTER_PORT"] = master_port
+    dist.init_process_group(backend=backend, rank=rank, world_size=world_size)
+
+
+def cleanup():
+    """Cleans up distributed backend resources."""
+    dist.destroy_process_group()
```

Comment:
We generally do not need to do this. Did you see any error without doing this?

Reply:
> A nit-picky comment - the example shows how to use torch.distributed (and not DDP). Maybe we can change the scope of the PR to focus on an example of torch.distributed? Alternatively, the ImageNet example will be easy to port here or maybe a simpler example, but with a model and dataloader etc.

I am also considering this this example belongs there.
A more complete example can live in https://github.com/pytorch/hydra-torch (which is still WIP).

Reply:
>  A nit-picky comment - the example shows how to use `torch.distributed` (and not DDP). Maybe we can change the scope of the PR to focus on an example of `torch.distributed`? Alternatively, the [ImageNet example](https://github.com/pytorch/examples/tree/master/imagenet) will be easy to port here or maybe a simpler example, but with a model and dataloader etc.
I completely agree with you. The reason why I implemented a `torch.distributed` example for now is to create a prototypical or minimal example that demonstrates multirun and Joblib being used to launch the distributed processes. I'm in the middle of fixing my GPU cluster, after which I plan to commit a full DDP example.

Reply:
>It is not guaranteed that all the jobs in JobLib will start at the same time (or around the same time). This is probably outside the scope of this PR but something to think about in general.

Is there a way to guarantee that the jobs start together?

> It is not guaranteed that the different processes will be scheduled on the same node. As Omry mentioned, it will be useful to pass the parent's ip (I think that means that the parent process participates in the rendezvous).

I intended this PR to be the single node case. I will definitely try to work on the multi-node case, but I'm less familiar with it compared to the single node case.

Reply:
@omry should I create PR in https://github.com/pytorch/hydra-torch instead for the ImageNet DDP example?

Reply:
@briankosw,
If you want to create a DDP example based on ImageNet - yes.
I also think that this example probably should go there as well. (hydra-torch repository did not exist when I first created the issue).